### PR TITLE
Conversion of DonateGuild and DonateMonthly

### LINKED ISSUE #33: Convert Donate guild and Donate Monthly to typescript

### PR DESCRIPTION
**Pull request by MohitPopli**
closes #33

### CODE REVIEW

File: components/forms/Donate/Donateguild.tsx [hunk 1]
```diff
@@ -28,25 +28,31 @@ export default function DonateGuild() {
...
+      if (stripe !== null) {
+        const { error } = await stripe.redirectToCheckout({
+          sessionId: response.id,
+        })
+        console.warn(error.message)
+      }
+      setLoading(false)
+    } catch (error) {
+      console.error(error)
+    }
```
Comment (mpeyper) [inline components/forms/Donate/Donateguild.tsx]: Did you need to introduce the `try`/`catch` for the conversion?

File: components/forms/Donate/Donateguild.tsx [hunk 2]
```diff
@@ -9,15 +9,15 @@ const stepThree = ' covers an entire month of therapy sessions.'
...
 
 export default function DonateGuild() {
-  const [loading, setLoading] = useState(false)
-  const [input, setInput] = useState()
-  const [impact, setImpact] = useState('$30')
-  const [message, setMessage] = useState(stepTwo)
+  const [loading, setLoading] = useState<boolean>(false)
+  const [input, setInput] = useState({ value: 0 })
+  const [impact, setImpact] = useState<string>('$30')
+  const [message, setMessage] = useState<string>(stepTwo)
```
Comment (mpeyper) [inline components/forms/Donate/Donateguild.tsx]: Bit of a nitpick, but I think all the generics here can be inferred

```suggestion
  const [loading, setLoading] = useState(false)
  const [input, setInput] = useState({ value: 0 })
  const [impact, setImpact] = useState('$30')
  const [message, setMessage] = useState(stepTwo)
```

File: components/forms/Donate/Donateguild.tsx (hunk 1 above)
Comment (JacobMGEvans) [reply components/forms/Donate/Donateguild.tsx]: Yeah, this is changing how the code originally functioned. Not necessary for the typescript conversín

Comment (MohitPopli) [reply components/forms/Donate/Donateguild.tsx]: Understood. Removed `try/catch` block. I was bit confused as in should I introduce them at this stage or not. Thanks :)

File: components/forms/Donate/Donateguild.tsx (hunk 2 above)
Comment (MohitPopli) [reply components/forms/Donate/Donateguild.tsx]: Implemented

File: components/forms/Donate/Donateguild.tsx [hunk 3]
```diff
@@ -28,7 +28,7 @@ export default function DonateGuild() {
 
   console.log(input)
```
Comment (mitchelvanbever) [inline components/forms/Donate/Donateguild.tsx:29]: should we remove this log?

File: components/forms/Donate/Donatemonthly.tsx [hunk 4]
```diff
@@ -30,7 +30,7 @@ export default function Donatemonthly() {
 
   console.log(input)
```
Comment (mitchelvanbever) [inline components/forms/Donate/Donatemonthly.tsx:31]: same here

File: components/forms/Donate/Donatemonthly.tsx [hunk 5]
```diff
@@ -11,14 +11,14 @@ const stepFour = ' covers the cost of four therapy sessions each month.'
 
 export default function Donatemonthly() {
   const [loading, setLoading] = useState(false)
-  const [input, setInput] = useState()
+  const [input, setInput] = useState({ value: 0 })
```
Comment (mitchelvanbever) [inline components/forms/Donate/Donatemonthly.tsx:14]: also not sure if we should be changing the initial state here

File: components/forms/Donate/Donateguild.tsx [hunk 6]
```diff
@@ -10,14 +10,14 @@ const stepFour = ' covers an entire month of therapy sessions for two people.'
 
 export default function DonateGuild() {
   const [loading, setLoading] = useState(false)
-  const [input, setInput] = useState()
+  const [input, setInput] = useState({ value: 0 })
```
Comment (mitchelvanbever) [inline components/forms/Donate/Donateguild.tsx:13]: also not sure if we should be changing the initial state here

File: components/forms/Donate/Donatemonthly.tsx [hunk 7]
```diff
@@ -11,14 +11,14 @@ const stepFour = ' covers the cost of four therapy sessions each month.'
...
-  const handleInputChange = (e) => {
-    const id = e.target.id
-    const value = e.target.value
-    const provides = e.target.step
+  const handleInputChange = (e: FormEvent<HTMLInputElement>) => {
+    const id = Number(e.currentTarget.id)
+    const value = Number(e.currentTarget.value)
+    const provides = e.currentTarget.step
 
     setInput({
```
Comment (mitchelvanbever) [inline components/forms/Donate/Donatemonthly.tsx:23]: this spread also really doesn't make sense, but let's leave it as is.

File: components/forms/Donate/Donatemonthly.tsx [hunk 8]
```diff
@@ -11,14 +11,14 @@ const stepFour = ' covers the cost of four therapy sessions each month.'
...
   const [impact, setImpact] = useState('$30')
   const [message, setMessage] = useState(stepTwo)
 
-  const handleInputChange = (e) => {
-    const id = e.target.id
-    const value = e.target.value
-    const provides = e.target.step
+  const handleInputChange = (e: FormEvent<HTMLInputElement>) => {
+    const id = Number(e.currentTarget.id)
+    const value = Number(e.currentTarget.value)
```
Comment (mitchelvanbever) [inline components/forms/Donate/Donatemonthly.tsx]: Is there a reason to go with the generic Number casting vs parseInt or parseFloat. I believe you'll always receive a string and stripe will probably either want an Int or a Float

File: components/forms/Donate/Donateguild.tsx (hunk 3 above)
Comment (JacobMGEvans) [reply components/forms/Donate/Donateguild.tsx:29]: It was there before. We can get the TS conversion in then make other improvements later.

File: components/forms/Donate/Donatemonthly.tsx (hunk 7 above)
Comment (JacobMGEvans) [reply components/forms/Donate/Donatemonthly.tsx:23]: Same feels when I did it eariler

File: components/forms/Donate/Donatemonthly.tsx (hunk 8 above)
Comment (JacobMGEvans) [reply components/forms/Donate/Donatemonthly.tsx]: The donations, I think, are not floats but I think it should at least be `parseInt()`

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]
Number() -> parseInt()

File: components/forms/Donate/Donatemonthly.tsx [hunk 9]
```diff
@@ -11,14 +11,14 @@ const stepFour = ' covers the cost of four therapy sessions each month.'
...
+  const [input, setInput] = useState({ value: 0 })
   const [impact, setImpact] = useState('$30')
   const [message, setMessage] = useState(stepTwo)
 
-  const handleInputChange = (e) => {
-    const id = e.target.id
-    const value = e.target.value
-    const provides = e.target.step
+  const handleInputChange = (e: FormEvent<HTMLInputElement>) => {
+    const id = Number(e.currentTarget.id)
```
Comment (JacobMGEvans) [inline components/forms/Donate/Donatemonthly.tsx]:
```suggestion
    const id = parseInt(e.currentTarget.id)
```

File: components/forms/Donate/Donatemonthly.tsx (hunk 8 above)
Comment (JacobMGEvans) [reply components/forms/Donate/Donatemonthly.tsx]:
```suggestion
    const value = parseInt(e.currentTarget.value)
```

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]

**Pull request merged**